File: FrontEnd/food_order/lib/view_models/SaleCampaigns/SaleCampaignVM.dart
import 'package:food_delivery/view_models/Foods/FoodVM.dart';
import 'package:json_annotation/json_annotation.dart';

part 'SaleCampaignVM.g.dart';

@JsonSerializable()
class SaleCampaignVM {
  SaleCampaignVM();
  late int id;
  late String name;
  late String desciption;
  late DateTime startDate;
  late DateTime endDate;
  late double percent;
  late bool enabled;
  late int priority;

  late List<FoodVM>? foodVMs;

  factory SaleCampaignVM.fromJson(Map<String, dynamic> json) =>
      _$SaleCampaignVMFromJson(json);
  Map<String, dynamic> toJson() => _$SaleCampaignVMToJson(this);
}

File: BackEnd/FoodOrder.Core/ViewModels/SaleCampaigns/SaleCampaignVM.cs
using FoodOrder.Core.ViewModels.Foods;
using FoodOrder.Core.ViewModels.Orders;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FoodOrder.Core.ViewModels.SaleCampaigns
{
    public class SaleCampaignVM
    {
        public int ID { get; set; }
        public string Name { get; set; }
        public string Desciption { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public float Percent { get; set; }
        public bool Enabled { get; set; }
        public int Priority { get; set; }

        public List<OrderVM> OrderVMs { get; set; }
        public List<FoodVM> FoodVMs { get; set; }
    }
}
